diff --git a/node_modules/@portaljs/remark-wiki-link/dist/index.cjs b/node_modules/@portaljs/remark-wiki-link/dist/index.cjs
index 80d21bb..c836920 100644
--- a/node_modules/@portaljs/remark-wiki-link/dist/index.cjs
+++ b/node_modules/@portaljs/remark-wiki-link/dist/index.cjs
@@ -366,11 +366,13 @@ function fromMarkdown() {
     current.data.alias = alias;
   }
   function exitWikiLink(token) {
-    var wikiLink = this.exit(token);
+    // https://github.com/datopian/datahub/issues/1059
+    var wikiLink = top(this.stack)
     var _wikiLink$data = wikiLink.data,
       isEmbed = _wikiLink$data.isEmbed,
       target = _wikiLink$data.target,
       alias = _wikiLink$data.alias;
+    this.exit(token);
     // eslint-disable-next-line no-useless-escape
     var wikiLinkWithHeadingPattern = /^((?:[\0-\t\x0B\f\x0E-\u2027\u202A-\uD7FF\uE000-\uFFFF]|[\uD800-\uDBFF][\uDC00-\uDFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF])*?)(#(?:[\0-\t\x0B\f\x0E-\u2027\u202A-\uD7FF\uE000-\uFFFF]|[\uD800-\uDBFF][\uDC00-\uDFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF])*)?$/;
     var _target$match = target.match(wikiLinkWithHeadingPattern),
diff --git a/node_modules/@portaljs/remark-wiki-link/dist/index.js b/node_modules/@portaljs/remark-wiki-link/dist/index.js
index 8cee892..8627df6 100644
--- a/node_modules/@portaljs/remark-wiki-link/dist/index.js
+++ b/node_modules/@portaljs/remark-wiki-link/dist/index.js
@@ -357,11 +357,13 @@ function fromMarkdown() {
     current.data.alias = alias;
   }
   function exitWikiLink(token) {
-    var wikiLink = this.exit(token);
+    // https://github.com/datopian/datahub/issues/1059
+    var wikiLink = top(this.stack);
     var _wikiLink$data = wikiLink.data,
       isEmbed = _wikiLink$data.isEmbed,
       target = _wikiLink$data.target,
       alias = _wikiLink$data.alias;
+    this.exit(token);
     // eslint-disable-next-line no-useless-escape
     var wikiLinkWithHeadingPattern = /^((?:[\0-\t\x0B\f\x0E-\u2027\u202A-\uD7FF\uE000-\uFFFF]|[\uD800-\uDBFF][\uDC00-\uDFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF])*?)(#(?:[\0-\t\x0B\f\x0E-\u2027\u202A-\uD7FF\uE000-\uFFFF]|[\uD800-\uDBFF][\uDC00-\uDFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF])*)?$/;
     var _target$match = target.match(wikiLinkWithHeadingPattern),
